packages/ai-jsx/src/core/constrained-output.tsx

```diff
@@ -0,0 +1,141 @@
+/**
+ * This module defines affordances for constraining the output of the model
+ * into specific formats, such as JSON, YAML, or Markdown.
+ */
+import * as LLMx from '../index.js';
+import { ChatCompletion, SystemMessage, AssistantMessage, UserMessage } from './completion';
+import { memo } from './memoize';
+import yaml from 'js-yaml';
+
+function isJsonString(str: string): boolean | string {
```

Comment:
This is a bit dangerous because someone could see it and write:

```
if (isJsonString(myStr)) { 
```

But this method will return a truthy value in both cases. 

Instead, I'd make the return type something like `{ isJsonString: true; error: null } | { isJsonString: false; error: string}`.